docs/proposals/policy.md

```diff
@@ -0,0 +1,264 @@
+ 1.  A Policy is a container object that attempts to mitigate referential integrity problems.  Referential integrity problems **still exist!**  Local RoleBindings can still reference global Roles stored in a different object, so dangling references can still happen.  In usage, a RoleBinding that references a missing Role will produce an error.
+ 1.  Roles versus Teams.  A Role is a grouping of PolicyRules.  A team is a grouping of users.  A RoleBinding associates a Role with a set of users and/or teams.  This distinction makes Teams easier to share across projects.
+
+Policy Evaluation
+In order to determine whether a request is authorized, the AuthorizationAttributes are tested in the following order:
+  1. all deny RoleBinding PolicyRules in the global namespace
+  1. all allow RoleBinding PolicyRules in the global namespace
+  1. all deny RoleBinding PolicyRules in the namespace
+  1. all allow RoleBinding PolicyRules in the namespace
+
+### Configtime Authorization Types
+These are the types used in the policy example above.  They allow us: to quickly find the policy rules that apply to namespace in etcd, to easily delegate project level control to a project admin, and to easily segregate all role control (a project admin can create/update/delete RoleBindings, but not Roles).
+```
+// PolicyRule holds information that describes a policy rule, but does not contain information 
+// about who the rule applies to or which namespace the rule applies to.
+type PolicyRule struct {
+	// Deny is true if any request matching this rule should be denied.  If false, any request matching this rule is allowed.
+	Deny bool
+	// Verbs is a list of Verbs that apply to ALL the ResourceKinds and AttributeRestrictions contained in this rule.  "*" represents all kinds.
+	Verbs []Verb
+	// AttributeRestrictions will vary depending on what the Authorizer/AuthorizationAttributeBuilder pair supports.  
+	AttributeRestrictions runtime.EmbeddedObject
+	// ResourceKinds is a list of kinds this rule applies to.  "*" represents all kinds.
+	ResourceKinds []string
+}
+
+// Role is a logical grouping of PolicyRules that can be referenced as a unit by RoleBindings.
+type Role struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	Rules []PolicyRule
+}
+
+// RoleBinding references a Role, but not contain it.  It adds who and namespace information.  
+// It can reference any Role in the same namespace or in the global namespace.
+type RoleBinding struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	UserNames    []string
+	TeamNames    []string
+
+	// Since Policy is a singleton, this is sufficient knowledge to locate a role
+	RoleRef kapi.ObjectReference
```

Comment:
> Should define behavior when RoleRef cannot be resolved. I think we would have to treat that like a deny.

No.  A missing role ref means that data integrity has been compromised.  That should be an error.  There's an ordering problem (you need to know everything all at once), but in the Runtime Types section, Authorizer.Authorize(..) (allowedBy string, deniedBecause string, error) to allow callers to distinguish between denied by lack of allow, denied by rule, and denied by failure.

I could go for (allow bool, allowedBy string, deniedBy string, error) for ease on callers if you wanted.